libs/docs/core/rtl-service/examples/rtl-service-basic-example/rtl-service-basic-example.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Direction } from '@angular/cdk/bidi';
+import { ChangeDetectionStrategy, Component, computed, effect, inject } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { ButtonComponent, FormLabelComponent, RtlService, SwitchComponent, TextComponent } from '@fundamental-ngx/core';
+
+@Component({
+    selector: 'fd-rtl-service-basic-example',
+    templateUrl: 'rtl-service-basic-example.component.html',
+    imports: [TextComponent, ButtonComponent, FormLabelComponent, SwitchComponent, FormsModule],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    standalone: true
+})
+export class RtlServiceBasicExampleComponent {
+    // Injecting the RtlService
+    private _rtlService = inject(RtlService);
+
+    // Example text
+    text = `
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
+    dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+    aliquip ex ea commodo consequat.
+
+    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore
+    eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia
+    deserunt mollit anim id est laborum.`;
+
+    // Signal to track RTL state
+    rtl$ = computed(() => !!this._rtlService?.rtlSignal());
+    // Signal to track a direction
+    _dir$ = computed<Direction>(() => {
+        return this.rtl$() ? 'rtl' : 'ltr';
+    });
+
+    constructor() {
+        effect(() => {
+            this._updateTextContainerDirection();
```

Comment:
I am new to the signals API (still doing most of my work in Angular 15), so correct me if wrong but doesn't an effect need to have a reference to a signal inside of it? What is the purpose of having "effect" here?